server/Controllers/SessionsController.cs

```diff
@@ -141,6 +141,22 @@ public async Task<ActionResult> UploadMap(Guid sessionId)
             return new OkResult();
         }
 
+        [HttpDelete("{sessionId}")]
+        public async Task<ActionResult> DeleteSession([FromRoute] Guid sessionId)
+        {
+            var sessionFromDb = await this.repository.GetByIdWithEvents(sessionId);
```

Comment:
technically for this check you could add a repository method that will only do count or something to not download all the data
(this is what my teammate db-nazi would recommend)

Reply:
done